diesel_codegen/src/util.rs

```diff
@@ -101,22 +101,29 @@ pub fn inner_of_option_ty(ty: &Ty) -> Option<&Ty> {
     }
 }
 
-pub fn get_options_from_input(attrs: &[Attribute], on_bug: fn() -> !)
-    -> Option<&[MetaItem]>
+pub fn get_options_from_input(attrs: &Vec<Attribute>, on_bug: fn() -> !)
+    -> Option<Vec<MetaItem>>
 {
     let options = attrs.iter().find(|a| a.name() == "options").map(|a| &a.value);
     match options {
-        Some(&MetaItem::List(_, ref options)) => Some(options),
+        Some(&MetaItem::List(_, ref options)) => {
+            Some(options.iter().map(|o| {
+                match o {
+                   &NestedMetaItem::MetaItem(ref m) => m.clone(),
+                   _ => on_bug(),
+                }
+            }).collect())
+        }
```

Comment:
This looks pretty complicated, cf. <https://github.com/diesel-rs/diesel/pull/508/files#r90534159>